### ISSUE TITLE
[noob] Unable to get the docker image working

### ISSUE THREAD

**Issue by nailyk-fr** [jitsi/docker-jitsi-meet#16]
Hi. 

Thanks for providing a docker image template. 
However, I am not able to make it working. 

# The setup: 
Running the image on a dedicated server with public ip on eth0. 
Having apache2 as proxy for both http(s) ports. 

After some `dock exec -it <id> cat/etc/passwd` I was able to set the right perms on some folder. For further references: 
```
root@khiksistigma :) # ls -aln 
total 32
drwxr-xr-x  8 1000 1000 4096 oct.   5 17:26 .
drwxrwxr-x 10 1000 1000 4096 oct.   5 16:26 ..
drwx------  2  999 1000 4096 oct.   5 16:51 jicofo
drwx------  2  999 1000 4096 oct.   5 17:38 jvb
drwxr-xr-x  4 1000 1000 4096 oct.   5 16:51 meet
drwxr-xr-x  5  101  102 4096 oct.   5 16:51 prosody
drwxr-xr-x  2  101  102 4096 oct.   5 17:31 prosody_var
drwxr-xr-x  4    0    0 4096 oct.   5 17:15 web
```
After reading other issues, am not running docker as root. 
[.env](https://github.com/jitsi/docker-jitsi-meet/files/2451518/env.txt)
[docker-compose.yml.txt](https://github.com/jitsi/docker-jitsi-meet/files/2451531/docker-compose.yml.txt)
As I had no idea which value tu setup on `DOCKER_HOST_ADDRESS` I put the public ip address. 

# the problem: 
After the `docker-compose up` everything seems to be working fine. Not so much errors in logs. Opening a webpage on jitsi.nailyk.fr seems to be working. However, when someone attempt to open the same link to join, the both webpages crashes and the log display `SEVERE: Can not invite participant -- no bridge available.`

Here is the list of opened ports: 
```
CONTAINER ID        IMAGE                       COMMAND                  CREATED             STATUS              PORTS                                                                                            NAMES
d97e2e1f4afa        jitsi/jvb                   "/init"                  5 seconds ago       Up 4 seconds        51.15.17.68:10000->10000/udp                                                                     docker-jitsi-meet_jvb_1_f812e25b8c76
a10386ce5886        jitsi/prosody               "/init"                  23 minutes ago      Up 3 seconds        5222/tcp, 5269/tcp, 5280-5281/tcp, 5347/tcp                                                      docker-jitsi-meet_prosody_1_740326e9447f
f5de03020013        jitsi/web                   "/init"                  33 minutes ago      Up 3 seconds        127.0.0.1:8000->80/tcp, 127.0.0.1:8443->443/tcp                                                  docker-jitsi-meet_web_1_1f78696201e1
7a11218fe595        jitsi/jicofo                "/init"                  About an hour ago   Up 4 seconds                                                                                                         docker-jitsi-meet_jicofo_1_a4c262113f25

[...]

udp        0      0 51.15.17.68:10000       0.0.0.0:*                           27351/docker-proxy  
```
so only 10000 is publicly exposed but doesn't seems to be open: 
```
root@khiksistigma 1 :( # nc localhost 10000 -vvvv
localhost [127.0.0.1] 10000 (webmin) : Connection refused
 sent 0, rcvd 0
```

I already tried to delete the whole generated files (`${CONFIG}`)without any changes. 
Docker log from `docker-compose up` to two tabs on the same room: 
[docker.log](https://github.com/jitsi/docker-jitsi-meet/files/2451592/docker.log)

What am I missing please? 
Thanks in advance!

**Comment by nailyk-fr** [comment]
Thanks for your answer! 
I am not running docker as root...

Le 17 octobre 2018 12:21:17 GMT+02:00, "Saúl Ibarra Corretgé" <notifications@github.com> a écrit :
>Generally running as root is a bad idea because the containers drop
>privileges when they start. Can you try running docker-compose as a
>regular user?
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/jitsi/docker-jitsi-meet/issues/16#issuecomment-430574140

-- 
Envoyé de mon appareil LineageOS avec K-9 Mail. Veuillez excuser ma brièveté.

**Comment by saghul** [comment]
Does this machine have a public or private IP address? if it has a private one, set it on DOCKER_HOST_ADDRESS, otherwise there is no need to set it. And make sure there is no firewall interfering.

**Comment by nailyk-fr** [comment]
No change without the `DOCKER_HOST_ADDRESS`. 
It is a dedicated server with public address directly connected without firewall. `docker-compose.yml` (partial): 
```
    # Video bridge
    jvb:
        image: jitsi/jvb
        ports:
            - '51.15.17.68:${JVB_PORT}:${JVB_PORT}/udp'
```

**Comment by saghul** [comment]
Why did you modify the compose file? You generally don’t need to do that. Can you share the jicofo logs?

**Comment by nailyk-fr** [comment]
> Why did you modify the compose file? You generally don’t need to do that. Can you share the jicofo logs?

Cause I attempt almost everything to fix it :wink: (especially port redirections)
[jicofo.log.txt](https://github.com/jitsi/docker-jitsi-meet/files/2488559/jicofo.log.txt)

**Comment by saghul** [comment]
There is a “not authorized” error. Did you modify the password in the env file after you had already started the containers? If so the change won’t take effect.

Delete the directory you specified in CONFIG and run it again. That should recreate the config files and hopefully it works now!

**Comment by nailyk-fr** [comment]
Yeah, this error is present from the beginning. Some base folder (aka docker mount point) even set as user, are changed to root on every start. No way to pass that error. 
I tired once to change the password. I never did it again when I saw all the failures messages :)

**Comment by saghul** [comment]
> Some base folder (aka docker mount point) even set as user, are changed to root on every start.

That's ok and expected. Not an error.

Can you show the prosody logs?

**Comment by nailyk-fr** [comment]
Sorry for being such late :(

New try, same issue: 
[prosody.log](https://github.com/jitsi/docker-jitsi-meet/files/2502589/prosody.log)
[web.log](https://github.com/jitsi/docker-jitsi-meet/files/2502590/web.log)
[jicofo.log](https://github.com/jitsi/docker-jitsi-meet/files/2502591/jicofo.log)
JVB log cannot be attached as it is 40Mb ....

Thanks for taking care :)

**Comment by saghul** [comment]
Ok, I think I know what's happening. You seem to be running old images. I noticed an error creating the `/config/certs/` directory (in the prosody logs) which was fixed over 2 months ago. So, let's try the following:

Stop and remove all your containers:

```
docker-compose stop
docker-compose rm
docker network prune
```

Remove the directory you specified in CONFIG.

Now let's fetch the new images:

```
docker-compose pull
```

Re-create the env file:

```
cp env.example .env
```

Customize it to your liking.

Start the containers again:

```
docker-compose up
```

I hope it works now!

**Comment by nailyk-fr** [comment]
Sadly.... Same issue. 

rm'ed everything to start from scratch: 

```
nailyk@khiksistigma :) % LC_ALL=C git status                                                                                                                                                                                              [master] ??~/docker/docker-jitsi-meet
On branch master
Your branch is up-to-date with 'origin/master'.
Untracked files:
  (use "git add <file>..." to include in what will be committed)

        docker-compose

nothing added to commit but untracked files present (use "git add" to track)
```
```
nailyk@khiksistigma :) % cat .env                                                                                                                                                           [master] ??~/docker/docker-jitsi-meet
# Directory where all configuration will be stored.
CONFIG=/media/datas/dockers/jitsi

# Exposed HTTP port.
HTTP_PORT=8000

# Exposed HTTPS port.
HTTPS_PORT=8443

# System time zone.
TZ=Europe/Amsterdam

# Internal XMPP domain. You generally don't need to change this.
XMPP_DOMAIN=meet.jitsi

# Internal XMPP domain for authenticated services. You generally don't need to
# change this.
XMPP_AUTH_DOMAIN=auth.meet.jitsi

# XMPP BOSH URL base. You generally don't need to change this.
XMPP_BOSH_URL_BASE=http://xmpp.meet.jitsi:5280

# XMPP domain for the MUC. You generally don't need to change this.
XMPP_MUC_DOMAIN=muc.meet.jitsi

# XMPP component password for Jitsi Videobridge.
JVB_COMPONENT_SECRET=s3cr3t

# STUN servers used to discover the server's public IP.
JVB_STUN_SERVERS=stun.l.google.com:19302,stun1.l.google.com:19302,stun2.l.google.com:19302

# Media port for the Jitsi Videobridge
JVB_PORT=10000

# XMPP component password for Jicofo.
JICOFO_COMPONENT_SECRET=s3cr37

# XMPP user for Jicofo client connections. You generally don't need to change
# this. NOTE: this option doesn't currently work due to a bug.
JICOFO_AUTH_USER=focus

# XMPP password for Jicofo client connections.
JICOFO_AUTH_PASSWORD=passw0rd

# IP address of the Docker host. See the "Running on a LAN environment" section
# in the README.
#DOCKER_HOST_ADDRESS=192.168.1.1
```
```
nailyk@khiksistigma :) % ls -ln /media/datas/dockers/jitsi                                                                                                                                                                   /tmp
total 16
drwx------ 2  999 1000 4096 oct.  24 15:52 jicofo
drwx------ 2  999 1000 4096 oct.  24 15:52 jvb
drwxr-xr-x 4 1000 1000 4096 oct.  24 15:52 meet
drwxr-xr-x 5  101  102 4096 oct.  24 15:52 prosody
```

**logs:** 
[web.log](https://github.com/jitsi/docker-jitsi-meet/files/2511310/web.log)
[jicofo.log](https://github.com/jitsi/docker-jitsi-meet/files/2511311/jicofo.log)
[jvb.log](https://github.com/jitsi/docker-jitsi-meet/files/2511312/jvb.log)
[prosody.log](https://github.com/jitsi/docker-jitsi-meet/files/2511313/prosody.log)

Sorry :(

**Comment by saghul** [comment]
I noticed 2 things: first, you haven't followed my instructions above because your `.env` file is old. **Please** start with a fresh copy.

Then your prosody container gets SIGHUP and reloads the config, but there are no values there anymore and the certs are gone. This is very weird. Can you please paste **exactly** what commands you are typing?

**Comment by nailyk-fr** [comment]
Sorry about the `env` file...  I did a diff between `env.exemple` and mine and noticed only the CONFIG folder being different. 
```
nailyk@khiksistigma :) % diff -upN env.example .env                                                                                                                                         
[master] ??~/docker/docker-jitsi-meet
--- env.example 2018-09-29 10:02:51.211968225 +0000
+++ .env        2018-10-24 15:18:55.009260471 +0000
@@ -1,5 +1,5 @@
 # Directory where all configuration will be stored.
-CONFIG=~/.jitsi-meet-cfg
+CONFIG=/media/datas/dockers/jitsi

 # Exposed HTTP port.
 HTTP_PORT=8000
```

Basically did all the command you mentioned, :
![capture d ecran_2018-10-24_21-43-41](https://user-images.githubusercontent.com/16372405/47456853-db2d8700-d7d5-11e8-8180-f0d65287bff6.png)
```
./docker-compose rm
docker network prune
./docker-compose pull
diff -upN env.example .env
sudo rm -rf /media/datas/dockers/jitsi/*/*
```
Then I had the same permission problem as before so attempt to fix them with the same procedure as in the first post of this issue. 
Then attempt to copy my lets encrypt certificate but it failed so I removed it again. 

Sorry having missed something. 
I will try again tomorow with a `git pull` too to be sure starting from scratch this time. 
(`./docker-compose` because I have the binary into the git folder)

**Comment by saghul** [comment]
Please do a `git pull` first. Also, what are you doing with some certs? Those are only used internally and you shouldn't replace them.

**Comment by saghul** [comment]
Also, check the permissions on your config folder, starting from the root.

**Comment by nailyk-fr** [comment]
ok....
```
 1060  rm -rf docker-jitsi-meet
 1061  sudo rm -rf /media/datas/dockers/jitsi
 1062  git clone https://github.com/jitsi/docker-jitsi-meet.git
 1063  cd docker-jitsi-meet
 1064  wget "https://github.com/docker/compose/releases/download/1.23.0-rc3/docker-compose-Linux-x86_64" -O docker-compose
 1065  ./docker-compose --help
 1066  chmod +x docker-compose
 1067  ./docker-compose --help
 1068  cp env.example .env
 1069  vim .env (changed config to /tmp/jitsi to ensure 777 mode)
 1070  mkdir /tmp/jitsi -m 777
 1071  ll /tmp/
 1072  ./docker-compose pull
 1073  ll /
 1074  ./docker-compose up
```

**TL;DR;** removed everything and start from scratch. 
Now working fine. Cannot get it. 
Many thanks for your help, I will try to reproduce the issue this week-end.

**Comment by nailyk-fr** [comment]
getting back the repo to 7dbbc75fedfcd7ed1e52ea3e9e60c52df30eae26 make it crash again ;)
I can do a dichotomy if you'd like but as you fixed the issues, guess it is not needed anymore :)
Many thanks!

**Comment by saghul** [comment]
Glad it works for you! I don't think that commit had anything to do with the failures, so better spend the time playing with your now working setup! :-)

**Comment by nailyk-fr** [comment]
Sorry, meant, if I rollback the git tree to that commit, I encounter the issue again. 
Need to try another point too about 777 chmod on the ${CONFIG} folder.
